### ISSUE TITLE
It would be helpful to add a tip for reusable/composable function

### ISSUE THREAD

**Issue by homura** [remeda/remeda#864]
I have noticed that Remeda is extremely TypeScript-friendly. However, I have found that most of the documentation uses the function `pipe` to demonstrate the **_data-last_** approach. Although there is some information about it in the description of the `piped` function:

https://github.com/remeda/remeda/blob/6dbc5655481121afe44d8d0e9625b27c46de2a83/src/piped.ts#L10-L11

I think that this information is not prominent enough. It would be helpful to include a tip in the guide that demonstrates how to create a type-friendly reusable function for the **_data-last_** approach. For instance:

```ts
const youngest = (gender: 'm' | 'f') =>
  R.piped(
    // to help type inference
    (users: User[]) => users, // or use the R.identity()<User[]>
    R.filter((x) => x.gender === gender),
    R.firstBy([R.prop('age'), 'asc']),
  );

const yongestM = youngest('m')(DATA);
const yangestF = youngest('f')(DATA);
```

**Comment by eranhirsch** [comment]
Until we can deprecate versions of typescript before 5.4 we can't use `NoInfer` in our types. Without it, a lot of functions, `piped` included, can't be typed correctly to handle both types that are inferred from the call site (like in pipes), and types which are inferred from a future invocation (like in "naked" data-last calls). For that reason we don't recommend using our utilities this way.

The main use-case for piped is not to react helper utils, but for cases where you'd need more complex logic that could be composed from other utils *as a callback* to another function. e.g.

```ts
const result = pipe(
  DATA,
  groupBy(...),
  mapValues(piped(filter(...), sortBy(...), map(...), first())
);
```

**Comment by tobychidi** [comment]
This is an important question because I want to implement a kind of selective serialization. The only way I have got it to work while being type-safe so far is with `map` and `evolve`. I spent a long time trying to write a serializer. 


This is what I have now

```js

const serialized = map(scheduleRecords, (record) =>
      evolve(record, {
         schedules: (schedules) =>
            map(schedules, (schedule) =>
               evolve(schedule, {
                  startDate: (date) => mSerialize(date),
                  endDate: (date) => (date ? mSerialize(date) : null),
               })
            ),
         holidays: (holidays) =>
            map(holidays, (holiday) =>
               evolve(holiday, {
                  startDate: (date) => mSerialize(date),
                  endDate: (date) => mSerialize(date),
               })
            ),
         leaves: (leaves) =>
            map(leaves, (leave) =>
               evolve(leave, {
                  startDate: (date) => mSerialize(date),
                  endDate: (date) => mSerialize(date),
               })
            ),
         suspensions: (suspensions) =>
            map(suspensions, (suspension) =>
               evolve(suspension, {
                  startDate: (date) => mSerialize(date),
                  endDate: (date) => mSerialize(date),
               })
            ),
      })
   );
   
   ```
   
 1. My attempts to create a reusable composable for this to make serialization and deserialization reuse the same shape have failed. The extended generic types as won't work, especially for evolve.
 
 
2. Ideally, I was intially thinking of creating a transform function that works with any data shape while still being type-safe. 
 
  
 Any of these two solutions work for me but I think the first on is more performant. Any advice on this issue?

**Comment by cjquines** [comment]
yeah; we've definitely been considering more utilities for deep arbitrary objects. https://github.com/remeda/remeda/issues/170 and such

**Comment by eranhirsch** [comment]
I don't feel that using `piped` to create data-last functions is something we want to emphasize in the function's documentation. This could be a great blog article, and we can consider linking to such a document from within the documentation or even hosting an article on the documentation site by creating a new section for longer-form articles and guides.

As it stands right now, this issue doesn't have anything actionable we can do to resolve it. Feel free to re-open the issue if anyone feels otherwise. Another idea is to move it to our repo's "Discussions" section.
